packages/app/src/cli/services/environment/identifiers-extensions.ts

```diff
@@ -37,6 +37,20 @@ export async function ensureExtensionsIds(
     onlyRemoteExtensions = matchResult.onlyRemote
   }
 
+  for (const extension of matchExtensions.toMigrate) {
+    // eslint-disable-next-line no-await-in-loop
+    const confirmed = await migratePrompt(extension.local, extension.remote)
+
+    if (!confirmed) {
+      return err('user-cancelled')
+    }
+
+    // eslint-disable-next-line no-await-in-loop
+    await migrateToUiExtension(extension.remote.id, options.appId)
+
+    output.completed(`Migrated ${extension.local.configuration.name}.`)
+  }
```

Comment:
I think it will be better to have 1 prompt asking for all migrations at the same time than doing it sequentially, the deploy will be cancelled anyway if you don't accept all migrations. // @MeredithCastile 
Same thing we do right now when we ask the user to confirm adding/updating an extension

Reply:
Yeah, fair.  Lemme spike this.

Reply:
Any strong preferences @MeredithCastile @lemonmade @isaacroldan ?

You can checkout the last 2 commits and then test each experience using the tophat instructions in the PR description.

Reply:
Looking at the last prompt I found it a bit confusing that it's saying it's going to update and permanently migrate the same `checkout-ui` extension.

Also that confirmation prompt is the one that users can skip by passing the `--force` flag, so I'm not sure it's the best place to display vital information that we really want developers to see. My proposal would be something like this:

```
╭─ warning ────────────────────────────────────────────────────────────────────╮
│                                                                              │
│  Extension migration detected.                                               │
│                                                                              │
│  A migration is permanent and can't be undone. All previous versions will    │
│  be migrated but current users won't be affected.                            │
│                                                                              │
│  Reference                                                                   │
│    • Migration guide ( https://shopify.dev/tutorials/migrate-extensions )    │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯

? Migrate these extensions in Shopify Partners?

      checkout-ui: • from 'checkout_ui_extension'
                   • to 'ui_extension'

> (y) Yes, migrate these extensions
  (n) No, cancel

? Make the following changes to your extensions in Shopify Partners?

      Update:  • checkout-ui

> (y) Yes, deploy to push changes
  (n) No, cancel
```

In the above example, the prompt about migrating extensions would show up even if the user passes `--force` to the deploy command.

Reply:
Resolving this as it's now outdated.  Please see: https://github.com/Shopify/cli/pull/1179#issuecomment-1402144124 instead